spinalcordtoolbox/vertebrae/core.py

```diff
@@ -308,36 +309,50 @@ def get_z_and_disc_values_from_label(fname_label):
     return [z_label, value_label]
 
 
-def clean_labeled_segmentation(fname_labeled_seg, fname_seg, fname_labeled_seg_new):
+def clean_extra_labels(fname_labeled_seg, fname_seg):
     """
-    FIXME doc
-    Clean labeled segmentation by:\
-      (i)  removing voxels in segmentation_labeled that are not in segmentation and\
-      (ii) adding voxels in segmentation that are not in segmentation_labeled
-
-    :param fname_labeled_seg:
-    :param fname_seg:
-    :param fname_labeled_seg_new: output
-    :return: none
+    Remove voxels in labeled segmentation that are not in segmentation,
+    saving the results to the original labeled segmentation file.
+
+    :param str fname_labeled_seg: filename of labeled segmentation
+    :param str fname_seg: filename of segmentation
+    :return: None
     """
-    # remove voxels in segmentation_labeled that are not in segmentation
     img_labeled_seg = Image(fname_labeled_seg)
     img_seg = Image(fname_seg)
-    data_labeled_seg_mul = img_labeled_seg.data * img_seg.data
-    # dilate to add voxels in segmentation that are not in segmentation_labeled
-    data_labeled_seg_dil = dilate(img_labeled_seg.data, 2, 'ball')
-    data_labeled_seg_mul_bin = data_labeled_seg_mul > 0
-    data_diff = img_seg.data - data_labeled_seg_mul_bin
-    ind_nonzero = np.where(data_diff)
-    img_labeled_seg_corr = img_labeled_seg.copy()
-    img_labeled_seg_corr.data = data_labeled_seg_mul
-    for i_vox in range(len(ind_nonzero[0])):
-        # assign closest label value for this voxel
-        ix, iy, iz = ind_nonzero[0][i_vox], ind_nonzero[1][i_vox], ind_nonzero[2][i_vox]
-        img_labeled_seg_corr.data[ix, iy, iz] = data_labeled_seg_dil[ix, iy, iz]
-    # save new label file (overwrite)
-    img_labeled_seg_corr.absolutepath = fname_labeled_seg_new
-    img_labeled_seg_corr.save()
+    # img_seg voxel values are either 0 or 1
+    img_labeled_seg.data *= img_seg.data
+    img_labeled_seg.save()
```

Comment:
I might be blanking here, but given the refactor into `expand_labels` and `crop_labels`, do we still need this old `clean_extra_labels` function?

Reply:
Thanks for catching this! I just forgot to remove the function after rewriting it. Fixed now.